fem/transfer.cpp

```diff
@@ -1138,37 +1025,15 @@ void L2ProjectionGridTransfer::L2ProjectionL2Space::ProlongateTranspose(
 void L2ProjectionGridTransfer::L2ProjectionL2Space::EAProlongateTranspose(
    const Vector &x, Vector &y) const
 {
-   const int vdim = fes_ho.GetVDim();
-
    const int iho = 0;
    const int nref = ho2lor.RowSize(iho);
    const int ndof_ho = fes_ho.GetFE(iho)->GetDof();
    const int ndof_lor = fes_lor.GetFE(ho2lor.GetRow(iho)[0])->GetDof();
-   const Mesh *mesh_ho = fes_ho.GetMesh();
-   const int nel_ho = mesh_ho->GetNE();
-
-   auto v_P = Reshape(P.Read(), ndof_ho, ndof_lor, nref, nel_ho);
-   auto v_x    = Reshape(x.Read(), ndof_ho, vdim, nel_ho);
-   auto v_y    = Reshape(y.Write(), ndof_lor, nref, vdim, nel_ho);
+   const int nel_ho = fes_ho.GetMesh()->GetNE();
 
-   mfem::forall_3D(nel_ho, ndof_lor, nref, vdim, [=] MFEM_HOST_DEVICE (int e)
-   {
-      MFEM_FOREACH_THREAD(v, z, vdim)
-      {
-         MFEM_FOREACH_THREAD(iref, y, nref)
-         {
-            MFEM_FOREACH_THREAD(ilo, x, ndof_lor)
-            {
-               real_t dot = 0.0;
-               for (int iho=0; iho<ndof_ho; ++iho)
-               {
-                  dot += v_P(iho, ilo, iref, e) * v_x(iho, v, e);
-               }
-               v_y(ilo, iref, v, e) = dot;
-            }
-         }
-      }
-   });
+   DenseTensor P_dt;
+   P_dt.NewMemoryAndSize(P.GetMemory(), ndof_ho, ndof_lor * nref, nel_ho, false);
+   BatchedLinAlg::MultTranspose(P_dt, x, y);
```

Comment:
I think we are missing a loop over vdim right?

Reply:
It should be automatic based on the sizes (that's the idea anyway)